cli/tools/jupyter/server.rs

```diff
@@ -410,87 +410,81 @@ impl JupyterServer {
       let exception_details = exception_details.unwrap();
 
       // Determine the exception value and name
-      let (value, name) = if let Some(exception) = &exception_details.exception {
-        // Determine the exception name based on class_name or default to "Uncaught"
-        let name = if let Some(class_name) = &exception.class_name {
-            format!("Uncaught {}", class_name)
-        } else {
-            "Unknown exception".to_string()
-        };
+      let (name, message, stack) =
+        if let Some(exception) = exception_details.exception {
+          let result = self
+            .repl_session
+            .call_function_on_args(
+              r#"
+          function(object) {
+            if (object instanceof Error) {
+              const name = "name" in object ? String(object.name) : "";
+              const message = "message" in object ? String(object.message) : "";
+              const stack = "stack" in object ? String(object.stack) : "";
+              return JSON.stringify({ name, message, stack });
+            } else {
+              const message = String(object);
+              return JSON.stringify({ name: "", message, stack: "" });
+            }
+          }
+        "#
```

Comment:
Ooooh slick, this is exactly what I wanted to have done. Do we ever write these inline functions in typescript or do any other linting or typing checks on them?

Reply:
We can probably start to move some of these into extension scripts at some point, which will give us access to typescript and a slightly less inconvenient development experience. I imagine we'll have a few of them around!